libr/include/r_io.h

```diff
@@ -97,36 +97,53 @@ typedef struct r_io_undo_t {
 typedef struct r_io_undo_w_t {
 	bool set;
 	ut64 off;
-	ut8 *o;   /* old data */
-	ut8 *n;   /* new data */
-	size_t len;  /* length */
+	ut8 *o; /* old data */
+	ut8 *n; /* new data */
+	size_t len; /* length */
 } RIOUndoWrite;
 
+// io cache
+typedef struct io_cache_item_t {
+	RInterval *tree_itv;
+	RInterval itv;
+	ut8 *data;
+	ut8 *odata; // is this a good idea?
+	bool written;
+} RIOCacheItem;
+
+typedef struct r_io_cache_layer_t {
+	RPVector *vec; // a vector of items
```

Comment:
we can change this to Vec now. And since it is a pointer, we can use the forward declare

Reply:
i would prefer to get all the tests passing before switching to another data structure, to not touch much this code, but yeah, thats the plan :)